app/src/main/java/io/neurolab/fragments/FocusVisualFragment.java

```diff
@@ -62,6 +72,42 @@
     private SpaceAnimationVisuals rocketAnimation;
     private View view;
 
+    private UsbSerialInterface.UsbReadCallback readCallback = new UsbSerialInterface.UsbReadCallback() { //Defining a Callback which triggers whenever data is read.
+        @Override
+        public void onReceivedData(byte[] arg0) {
+            String data = null;
+            try {
+                data = new String(arg0, "UTF-8");
+                FilePathUtil.recordData(data);
+            } catch (UnsupportedEncodingException e) {
+                e.printStackTrace();
+            }
+        }
+    };
+
+    private final BroadcastReceiver broadcastReceiver = new BroadcastReceiver() { //Broadcast Receiver to automatically start and stop the Serial connection.
```

Comment:
Why do we have to re-implement this code every time we read from USB? Have a look at how USB reads are kept in a single file and used everywhere from there. This is a bad practice.

Reply:
Okay will check.